--- sources-orig/build_triton_wheel.py	2024-08-23 15:55:05.939767713 -0500
+++ sources-fixed/build_triton_wheel.py	2024-08-23 15:55:32.011615930 -0500
@@ -44,6 +44,25 @@
         f.write(orig)
 
 
+# TODO: remove patch_setup_py() once we have a proper fix for https://github.com/triton-lang/triton/issues/4527
+def patch_setup_py(path: Path) -> None:
+    with open(path) as f:
+        orig = f.read()
+    try:
+        orig = check_and_replace(
+            orig,
+            "https://tritonlang.blob.core.windows.net/llvm-builds/",
+            "https://oaitriton.blob.core.windows.net/public/llvm-builds/",
+        )
+        with open(path, "w") as f:
+            f.write(orig)
+    except RuntimeError as e:
+        print(
+            f"Applying patch_setup_py() for llvm-build package failed: {e}.",
+            "If you are trying to build a newer version of Triton, you can ignore this.",
+        )
+
+
 def build_triton(
     *,
     version: str,
@@ -82,6 +101,9 @@
         else:
             check_call(["git", "checkout", commit_hash], cwd=triton_basedir)
 
+        # TODO: remove this and patch_setup_py() once we have a proper fix for https://github.com/triton-lang/triton/issues/4527
+        patch_setup_py(triton_pythondir / "setup.py")
+
         if build_conda:
             with open(triton_basedir / "meta.yaml", "w") as meta:
                 print(
